### PR TITLE
[bsp][stm32]添加SDMMC驱动,以区分SDIO与SDMMC

### PR DESCRIPTION
**Pull request by wdfk-prog**
## 拉取/合并请求描述：(PR description)

[
#### 为什么提交这份PR (why to submit this PR)
[bsp][stm32]添加SDMMC驱动,以区分SDIO与SDMMC 
详细情况看:https://github.com/RT-Thread/rt-thread/issues/7088

#### 你的解决方案是什么 (what is your solution)
STM32的drv_sdio驱动分为sdio驱动和sdmmc驱动

- H7,F7等SDMMC驱动仅加载drv_sdmmc文件,SDIO驱动加载drv_sdio文件
#### 在什么测试环境下测试通过 (what is the test environment)
- ART-PI ;STM32H750
- SD卡
image
image

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Review by mysterywolf — APPROVED** [review]

File: bsp/stm32/libraries/HAL_Drivers/drv_sdmmc.c [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+
+    return order;
+}
+
+/**
+  * @brief  This function wait sdio cmd completed.
+  * @param  sdio rthw_sdio
+  * @retval None
+  */
+static void rthw_sdio_wait_completed(struct rthw_sdio *sdio)
```
Comment (Guozhanxin) [inline bsp/stm32/libraries/HAL_Drivers/drv_sdmmc.c:142]: 里面的函数命名，感觉也要调整

Comment (wdfk-prog) [reply bsp/stm32/libraries/HAL_Drivers/drv_sdmmc.c:142]: https://github.com/RT-Thread/rt-thread/blob/master/bsp/stm32/libraries/HAL_Drivers/drv_sdio.c#L126
sdio里面也是这个名字.要调整的话,得一起看看改成啥?

Comment (Guozhanxin) [reply bsp/stm32/libraries/HAL_Drivers/drv_sdmmc.c:142]: sdio --》sdmmc 这样呢

Comment (wdfk-prog) [reply bsp/stm32/libraries/HAL_Drivers/drv_sdmmc.c:142]: ok，没问题

**Pull request merged**